frontend/src/components/common/SimpleTable.tsx

```diff
@@ -72,37 +73,68 @@ export default function SimpleTable(props: SimpleTableProps) {
     emptyMessage = null,
     errorMessage = null,
   } = props;
+  const location = useLocation();
+  const history = useHistory();
   const [page, setPage] = React.useState(0);
   const [currentData, setCurrentData] = React.useState(data);
   const rowsPerPageOptions = props.rowsPerPage || [5, 10, 50];
   const [rowsPerPage, setRowsPerPage] = React.useState(rowsPerPageOptions[0]);
   const classes = useTableStyle();
 
+  function addQuery(queryObj:{[key:string]: string}) {
+    const pathname = location.pathname;
+    const searchParams = new URLSearchParams(location.search);
+    for (const key in queryObj) {
+      const value = queryObj[key];
+      if (value) {
+        searchParams.set(key, value);
+      } else {
+        searchParams.delete(key);
+      }
+    }
+
+    history.push({
+      pathname: pathname,
+      search: searchParams.toString()
+    });
+  };
+
   function handleChangePage(_event: any, newPage: number) {
+    addQuery({'page': newPage.toString(), 'rowsPerPage': rowsPerPage.toString()});
     setPage(newPage);
   }
 
   function handleChangeRowsPerPage(event: any) {
+    addQuery({'page': page.toString(), 'rowsPerPage': event.target.value});
     setRowsPerPage(+event.target.value);
     setPage(0);
   }
 
   React.useEffect(() => {
-    if (currentData === data) {
-      if (page !== 0) {
-        setPage(0);
-      }
+    if (data === currentData) {
       return;
     }
-
     // If the currentData is not up to date and we are in the first page, then update
     // it directly. Otherwise it will require user's intervention.
     if ((!currentData || currentData.length === 0) || page === 0) {
       setCurrentData(data);
     }
   },
   // eslint-disable-next-line
-  [data, currentData]);
+  [data])
+
+  function refreshClickHandler(data : SimpleTableProps['data']){
+    addQuery({'page': '0', 'rowsPerPage': rowsPerPage.toString()});
+    setCurrentData(data);
+  }
+
+  React.useEffect(() => {
+    const params = new URLSearchParams(location.search);
+    setPage(parseInt(params.get('page') as string) || 0);
+    setRowsPerPage(parseInt(params.get('rowsPerPage') as string) || rowsPerPageOptions[0]);
```

Comment:
We should verify this against the data we have, because if we type page=2 rowsPerPage=500 or so, it will still set that in the table (here's an example with 50 on page 2):
![Screenshot from 2020-10-20 16-17-49](https://user-images.githubusercontent.com/1029635/96608098-e8ac7780-12f0-11eb-8442-09a973d9079c.png)